.github/workflows/master-e2e.yaml

```diff
@@ -203,21 +207,24 @@ jobs:
     needs: create-runner
     runs-on: ${{ needs.create-runner.outputs.uuid }}
     env:
-      TIMEOUT_SCALE: 3
       ARCH: amd64
       CERT_MANAGER_VERSION: ${{ inputs.cert-manager_version }}
       CLUSTER_NAME: ${{ inputs.cluster_name }}
       CLUSTER_NS: fleet-default
       CLUSTER_TYPE: ${{ inputs.cluster_type }}
-      # For K3s installation used to host Rancher Manager
+    # K3S / RKE2 flags to use for installation
       INSTALL_K3S_EXEC: ${{ inputs.k3s_flags }}
-      INSTALL_K3S_VERSION: ${{ inputs.upstream_k3s_version }}
       INSTALL_K3S_SKIP_ENABLE: true
+      INSTALL_K3S_VERSION: ${{ inputs.upstream_cluster_version }}
+      INSTALL_RKE2_VERSION: ${{ inputs.upstream_cluster_version }}
       K3S_KUBECONFIG_MODE: 0644
-      # For Rancher Manager
-      RANCHER_VERSION: ${{ inputs.rancher_version }}
+      # Distribution to use to host Rancher Manager (K3s or RKE2)
+      K8S_UPSTREAM_DISTRIBUTION: ${{ inputs.upstream_cluster_distribution }}
```

Comment:
Maybe this can be extract from `upstream_cluster_version` instead creating a new variable? As the version "number" contains `k3s` or `rke2`.